### ISSUE TITLE
@fundamental-ngx/platform - UI5 Web Components Adoption

### ISSUE THREAD

**Issue by fkolar** [SAP/fundamental-ngx#7133]
#### Is this a bug, enhancement, or feature request?
feature

#### Briefly describe your proposal.
We  must ensure UI/UX consistency and the greatest reuse possible across different SAP products and applications developed by different teams using different UI frameworks.  There are still redundancies and alignment efforts which can be reduced by establishing a broader common foundation for all UI technology options. 

The goal of the UI technologies is to enable all SAP products to contribute to a consistent user experience across the entire product portfolio. Just like we are able to contribute to this common Angular Library and we were able to split this Library last year to provide both common (core) & application(platform) centric feature, its time to push this to the next level in order to active true reusability. (Or try to achieve).

Reusable UI elements will be implemented as Web Components and thus ensure a framework-agnostic reuse across different UI technologies. Since there is an existing project which tries to address this and provide foundation of core  components called UI5 Web Component, we should limit the effort of maintaining existing platform internals and invest this energy towards UI5 Web Components and see how we can adopt it. 

Our consumers the Angular Application Developers expects high-quality UI components that they can drop into existing applications without extra overhead. Therefore we need to make sure we can maintain the standard as it is now or even higher. We need to ensure that:
 * Component signature of platform components is unchanged and we keep the same number of features
 * We keep building more richer set  enterprise components
 * It provides Internationalization and accessibility 
 * Simple and easy to use API and Signature hiding unnecessary complexity
 * Customizable
 * Code is clean and well documented 
 * All tests are passing 

### Plan

* Analyze and compare existing set of components both on platform side and ui5 web component side
* List all missing features for each component and what needs to be additionally implemented when adopting such UI5 WC.
* Create list of components which are ready and we can adopt 
* Create List of component which are not ready or completely missing. Create possible roadmap 
* Plan and deliver road map for adoption. 

#### Additional Details

In parallel prepare a plan for Common CSS Layer that can be adopted by both UI5 Web Components as well as Fundamental styles (out of scope of this issue) 

_It seems like we are moving the right direction as several major framework providers already moved to this direction such as: ionic framework, PrimeNG_

**Comment by droshev** [comment]
Once we make a plan how to work closer with the web components we will publish our plan
